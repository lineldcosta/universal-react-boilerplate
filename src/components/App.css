.app {
  text-size-adjust: none;
  font-family: helvetica, arial, sans-serif;
  line-height: 200%;
  padding: 6px 20px 30px;
}

.title {
  font-weight: 200;
}

.navigation {
  padding: 0;
  padding-bottom: 1em;
}

.navigation li {
  list-style: none;
  display: inline-block;
  margin-right: 1em;
}

.navigation li:last-child {
  margin: 0;
}

.navigationLink {
  color: palevioletred;
  text-decoration: none;
  padding: 0.5em;
  border-radius: 3px;
}

.activeLink {
  color: white;
  background: palevioletred;
}
