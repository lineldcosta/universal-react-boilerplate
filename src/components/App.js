/* @flow */
import React from 'react'
import { NavLink } from 'react-router-dom'
import { renderRoutes } from 'react-router-config'
import routes from '../routes/routes'
import styles from './App.css'

import type { Element } from 'react'

type LinkProps = {
  to: string,
  exact?: boolean,
  children?: Element<*>
}

const Link = ({ to, exact, children }: LinkProps) => (
  <li>
    <NavLink
      to={to}
      exact={exact}
      className={styles.navigationLink}
      activeClassName={styles.activeLink}>{ children }</NavLink>
  </li>
)

type Props = {
  title?: string
}

const App = ({ title }: Props) => (
  <div className={styles.app}>
    <h1 className={styles.title}>{ title }</h1>
    <ul className={styles.navigation}>
      <Link exact to="/">Home</Link>
      <Link to="/counter">Counter</Link>
      <Link to="/async">Async</Link>
    </ul>
    {renderRoutes(routes)}
  </div>
)

App.propTypes = {
  title: React.PropTypes.string.isRequired
}

App.defaultProps = {
  title: 'Hello Universal React'
}

export default App
