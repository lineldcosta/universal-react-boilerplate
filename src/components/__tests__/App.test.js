/* eslint-env jest */
import React from 'react'
import App from '../App'
import { shallow } from 'enzyme'

const setup = propOverrides => {
  const props = Object.assign({
    title: 'Hello, tests'
  }, propOverrides)

  const wrapper = shallow(<App { ...props } />)

  return {
    props,
    wrapper,
    title: wrapper.find('.title')
  }
}

it('renders correctly', () => {
  const { wrapper } = setup()
  expect(wrapper).toMatchSnapshot()
})

it('has the correct title', () => {
  const { title } = setup({ title: 'Test Title' })
  expect(title.text()).toEqual('Test Title')
})
