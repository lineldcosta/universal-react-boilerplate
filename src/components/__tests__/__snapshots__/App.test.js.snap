// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="app"
>
    <h1
        className="title"
    >
        Hello, tests
    </h1>
    <ul
        className="navigation"
    >
        <Link
            exact={true}
            to="/"
        >
            Home
        </Link>
        <Link
            to="/counter"
        >
            Counter
        </Link>
        <Link
            to="/async"
        >
            Async
        </Link>
    </ul>
    <Switch>
        <Route
            path="/counter"
            render={[Function]}
        />
        <Route
            path="/async"
            render={[Function]}
        />
    </Switch>
</div>,
  "nodes": Array [
    <div
      className="app"
>
      <h1
            className="title"
      >
            Hello, tests
      </h1>
      <ul
            className="navigation"
      >
            <Link
                  exact={true}
                  to="/"
            >
                  Home
            </Link>
            <Link
                  to="/counter"
            >
                  Counter
            </Link>
            <Link
                  to="/async"
            >
                  Async
            </Link>
      </ul>
      <Switch>
            <Route
                  path="/counter"
                  render={[Function]}
            />
            <Route
                  path="/async"
                  render={[Function]}
            />
      </Switch>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <App
        title="Hello, tests"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "title": "Hello, tests",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="app"
>
          <h1
                    className="title"
          >
                    Hello, tests
          </h1>
          <ul
                    className="navigation"
          >
                    <Link
                              exact={true}
                              to="/"
                    >
                              Home
                    </Link>
                    <Link
                              to="/counter"
                    >
                              Counter
                    </Link>
                    <Link
                              to="/async"
                    >
                              Async
                    </Link>
          </ul>
          <Switch>
                    <Route
                              path="/counter"
                              render={[Function]}
                    />
                    <Route
                              path="/async"
                              render={[Function]}
                    />
          </Switch>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="app"
>
          <h1
                    className="title"
          >
                    Hello, tests
          </h1>
          <ul
                    className="navigation"
          >
                    <Link
                              exact={true}
                              to="/"
                    >
                              Home
                    </Link>
                    <Link
                              to="/counter"
                    >
                              Counter
                    </Link>
                    <Link
                              to="/async"
                    >
                              Async
                    </Link>
          </ul>
          <Switch>
                    <Route
                              path="/counter"
                              render={[Function]}
                    />
                    <Route
                              path="/async"
                              render={[Function]}
                    />
          </Switch>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App
    title="Hello, tests"
/>,
}
`;
